### PR TITLE
Port Android Handle Manager tests to Android Entity Handles

### PR DESCRIPTION
**Pull request by jblebrun**

### CODE REVIEW

File: javatests/arcs/android/entity/DifferentAndroidHandleManagerDifferentStoresTest.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        )
+        // Initialize WorkManager for instrumentation tests.
+        WorkManagerTestInitHelper.initializeTestWorkManager(app)
+    }
+
+    // TODO - fix these?
+    override fun collection_referenceLiveness() {}
+    override fun singleton_referenceLiveness() {}
+
+    // We don't expect these to pass, since Operations won't make it through the driver level
```
Comment (jblebrun) [inline javatests/arcs/android/entity/DifferentAndroidHandleManagerDifferentStoresTest.kt]: @csilvestrini This statement is correct, right?

File: javatests/arcs/core/entity/BUILD [hunk 2]
```diff
@@ -43,6 +43,7 @@ arcs_kt_jvm_library(
         ["*.kt"],
         exclude = TEST_SRCS,
     ),
+    visibility = ["//javatests/arcs/android/entity:__pkg__"],
```
Comment (jblebrun) [inline javatests/arcs/core/entity/BUILD]: @csilvestrini Is this legit?

File: javatests/arcs/android/entity/BUILD [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    "arcs_kt_android_test_suite",
+)
+
+licenses(["notice"])
+
+TEST_SRCS = glob(["*Test.kt"])
+
+arcs_kt_android_test_suite(
+    name = "entity",
+    srcs = TEST_SRCS,
```
Comment (csilvestrini) [inline javatests/arcs/android/entity/BUILD]: Just delete this line and the TEST_SRCS variable. The default is to glob all kt files

File: javatests/arcs/android/entity/DifferentAndroidHandleManagerDifferentStoresTest.kt [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    lateinit var app: Application
+
+    @Before
+    fun setUp() {
+        app = ApplicationProvider.getApplicationContext()
+        val testConnectionFactory = TestConnectionFactory(app)
+        readHandleManager = EntityHandleManager(
+            arcId = "arcId",
+            hostId = "hostId",
+                time = TimeImpl(),
```
Comment (csilvestrini) [inline javatests/arcs/android/entity/DifferentAndroidHandleManagerDifferentStoresTest.kt]: Weird indent

File: javatests/arcs/android/entity/DifferentAndroidHandleManagerDifferentStoresTest.kt (hunk 1 above)
Comment (csilvestrini) [reply javatests/arcs/android/entity/DifferentAndroidHandleManagerDifferentStoresTest.kt]: I have no idea...

File: javatests/arcs/core/entity/BUILD (hunk 2 above)
Comment (csilvestrini) [reply javatests/arcs/core/entity/BUILD]: This is legit, but copybara might not like it. Let's wait and see...

**Pull request merged**